src/ops.rs

```diff
@@ -440,23 +436,50 @@ fn op_chdir(
   }()))
 }
 
-fn op_set_timeout(
+fn op_global_timer_stop(
   isolate: &Isolate,
   base: &msg::Base<'_>,
   data: libdeno::deno_buf,
 ) -> Box<Op> {
+  assert!(base.sync());
   assert_eq!(data.len(), 0);
-  let inner = base.inner_as_set_timeout().unwrap();
-  let val = inner.timeout();
-  let timeout_due = if val >= 0 {
-    Some(Instant::now() + Duration::from_millis(val as u64))
-  } else {
-    None
-  };
-  isolate.set_timeout_due(timeout_due);
+  let mut t = isolate.state.global_timer.lock().unwrap();
+  t.cancel();
```

Comment:
Two questions:
* Does this make the global timer future report a failure, or will it simply never complete at all?
* Does this take effect synchronously? That is: if the the tokio delay has fired in the interim, will cancelling it pre-empt the future from (successfully) completing?

Reply:
* The global timer's errors are ignored (actually it panics) it only reports completion.
* When you cancel, if an existing timer future exists, it will complete successfully. b
* The canceled timer's callback is not triggered synchronously. At some point in the future the response Buf from the completed future gets written to the shared buffer, than then shuffled over to JS for its promise to be resolved.
* Cancelation and Timeout are not distinguished currently - so whether the pending future is canceled or completes - it is the same.